lot 14 partie Reporting

Iandraina-etech opened the pull request:
added a Prescriber Reporting on Comores Module

Iandraina-etech committed:
Add prescriber report generation and template

Introduces a new report for listing prestations by prescriber, including counts and breakdowns by type, period, and health facility. Adds the report logic in models.py, registers it in report.py with appropriate permissions, and creates a placeholder template file.

Iandraina-etech committed:
Ensure report data is JSON serializable

Updated report_prescriber_query to convert integer values to strings for JSON compatibility and to serialize the final data before returning. This change ensures the report output is consistently JSON serializable and avoids type issues in downstream consumers.

Iandraina-etech committed:
Merge pull request #1 from Iandraina-etech/feature-arkeuo

Feature arkeuo

Iandraina-etech committed:
Update report permissions to 131215

Changed the permission code for all report definitions from '131214' to '131215' to reflect updated access requirements.

Iandraina-etech committed:
Merge pull request #2 from Iandraina-etech/feature-arkeuo

Update report permissions to 131215

mngoe commented on reportcomores/models.py:
```diff
@@ -319,6 +319,262 @@ class PrintedReportsHistory(models.Model):
+            date_from__gte=date_start,
+            date_to__lte=date_end,
+            validity_to__isnull=True,
+            status__in=[4, 8, 16]  # Claims validés/approuvés
```
Je ne suis pas sur de comprendre d'ou vient cette contrainte.

Sauf erreur de ma part, je ne pense pas que le RFC définis un filtre spécifique par statut de claims.

mngoe commented on reportcomores/models.py:
```diff
@@ -319,6 +319,262 @@ class PrintedReportsHistory(models.Model):
+                    services_data[key]["count"] += claim_service.qty_provided or 1
+                    
+                    # Calcul du montant (priorité : valuated > approved > adjusted > asked)
+                    amount = (claim_service.price_valuated or 
```
Le calcul du montant semble faux.
Vous n'intégrez pas le concept de quantité (provided ou adjusted) dans le calcul total du montant par service.

mngoe replied:
Ce calcul de montant ne fonctionnera pas sur les services complexe.
Les services complexes comprennent d'autres services et items et la méthode de calcul ne peut pas être celle la.

mngoe commented on reportcomores/models.py:
```diff
@@ -319,6 +319,262 @@ class PrintedReportsHistory(models.Model):
+                             claim_service.price_adjusted or 
+                             claim_service.price_asked or 0)
+                    services_data[key]["total_amount"] += amount
+                    total_prestations += claim_service.qty_provided or 1
```
La qty_provided n'est pas la seule valeur prise en charge au niveau des quantités vu qu'elle peut être ajusté dans le processus de review.

mngoe commented on reportcomores/models.py:
```diff
@@ -319,6 +319,262 @@ class PrintedReportsHistory(models.Model):
+        for month_key in sorted(monthly_data.keys()):
+            data = monthly_data[month_key]
+            # Conversion du format YYYY-MM vers nom de mois
+            month_names = {
```
Traduction en dur a éviter.
Ces éléments sont déjà surement présent dans les traductions de base du BE (ou il faudrait les rajouter)

mngoe commented on reportcomores/models.py:
```diff
@@ -319,6 +319,262 @@ class PrintedReportsHistory(models.Model):
+                service = claim_service.service
+                if service:
+                    # Type de service
+                    service_type = service.type if service.type else "Non spécifié"
```
Pas de traduction en dur svp.

mngoe commented on reportcomores/models.py:
```diff
@@ -319,6 +319,262 @@ class PrintedReportsHistory(models.Model):
+                if service:
+                    # Type de service
+                    service_type = service.type if service.type else "Non spécifié"
+                    service_category = service.category if service.category else "Non spécifié"
```
Pas de traduction en dur svp.

mngoe commented on reportcomores/models.py:
```diff
@@ -319,6 +319,262 @@ class PrintedReportsHistory(models.Model):
+    print("Rapport Par Prescripteur", kwargs)
+    
+    # Récupération des paramètres
+    prescriber_uuid = kwargs.get("prescriber_uuid")
```
La vous fixer un rapport par Prescripteur.

Selon ma compréhension ce n'est pas ce qui est attendu en terme de rapport.

L'AMG veut un rapport qui liste le nombre de prestations par prescripteur (incluant de la répartition).
Selon ma compréhension c'est un seul rapport pour tous les prescripteurs.

Pour éviter les incompréhensions , je vous invite à le demander sur le JIRA directement à l'AMG pour être sur de ce développement

mngoe reviewed (changes requested):
A priori vous n'avez pas fait de GroupBy par Type d'acte. Je ne suis pas sur que juste les afficher répondra au besoin fonctionnel posé dans le ticket.

Iandraina-etech committed:
changed it to make suits the oici

Iandraina-etech committed:
finalized the second report on the reporting

Iandraina-etech committed:
ajustement finaux rapport 1

Iandraina-etech committed:
finished the report 1

Iandraina-etech committed:
improvement in code logic and syntax

Iandraina-etech committed:
debug

Iandraina-etech committed:
Refactor FOSA prescriber report logic and template

Replaces the old FOSA prescriber report query and template with new versions, updating the report definition and Python query references. Adds new aggregation logic and summary fields to the report query, improves handling of filter parameters, and introduces a new report template file.

Iandraina-etech committed:
Remove unnecessary float casting in report query

Eliminated redundant float conversions for prescriber and claim totals in report_fosa_prescriber_query. This simplifies the code and ensures values are formatted consistently.

Iandraina-etech committed:
Merge pull request #3 from Iandraina-etech/feature-rapport1-prescripteur

Feature rapport1 prescripteur

Iandraina-etech committed:
Update report permissions to 131215

Changed the permission code for all report definitions from '131214' to '131215' to reflect updated access requirements.

Iandraina-etech committed:
Merge pull request #4 from Iandraina-etech/feature-rapport1-prescripteur

Update report permissions to 131215

Iandraina-etech committed:
Merge pull request #5 from Iandraina-etech/feature-arkeuo

Feature arkeuo
